src/java/org/apache/cassandra/cql3/QueryEvents.java

```diff
@@ -235,6 +246,15 @@ public void notifyPrepareFailure(@Nullable CQLStatement statement, String query,
         }
     }
 
+    private String possiblyObfuscateQuery(CQLStatement statement, String query)
```

Comment:
@sumanth-pasupuleti this is really it? Are we sure they are set / changed just in these cases? I am just asking, seems to be rather simple :) it is possible to get if it is DCL statement from `statement.auditContext()` something like:

```
private String possiblyObfuscateQuery(CQLStatement statement, String query)
{
   return statement.getAuditLogContext().auditLogEntryType.getCategory() == AuditLogEntryCategory.DCL ?
         queryObfuscator.obfuscate(query) : query;
}
```

This seems to me to be more "robust". However, `getAuditLogContext` is creating new object every time so I am not sure about this approach and maybe we are better off, from performance perspective, to do it like you do. I just want to be sure that we have not forgotten anything else because if we work with credentials in any other statement in the future (which is quite unlikely), they would start to leak if we dont update this code. I would rather cover whole DCL class then individual statements.

Reply:
@smiklosovic we can maybe do this:
replace "if (statement instanceof CreateRoleStatement || statement instanceof AlterRoleStatement)" with
"if (statement instanceof AuthenticationStatement)" and that would then cover all DCL statements without having to rely on AuditLog related code to determine if it is DCL. Let me know what you think, and I can make the change

Reply:
@smiklosovic updated this change

Reply:
@sumanth-pasupuleti we might just simplify into this

    return statement instanceof AuthenticationStatement ? password.obfuscated(query) : query